source/features/enable-file-links-in-compare-view.tsx

```diff
@@ -0,0 +1,83 @@
+
+	// This solution accounts for:
+	// - Branches with slashes in it
+	// - PRs opened from the default branch
+	const dropdown = event.delegateTarget.nextElementSibling!;
+	const viewFile = select<HTMLAnchorElement>('[data-ga-click^="View file"]', dropdown)!;
+	const headBranchUrl = select<HTMLAnchorElement>('.commit-ref.head-ref a')!.pathname;
+	const filepath = dropdown.closest<HTMLDivElement>('[data-path]')!.dataset.path;
+	viewFile.pathname = headBranchUrl + '/' + String(filepath);
+}
+
+function handleCompareMenuOpening(event: delegate.Event): void {
+	event.delegateTarget.classList.add('rgh-actionable-link'); // Mark this as processed
+
+	const dropdown = event.delegateTarget.nextElementSibling!;
+	const viewFile = select<HTMLAnchorElement>('[data-ga-click^="View file"]', dropdown)!;
+	const url = new GitHubURL(viewFile.href);
+	url.assign({
+		branch: location.pathname.replace(/.+:|.+\.{3}/, '')
+	});
+	viewFile.href = String(url);
+
+	// Fix the edit link
+	const editFile = viewFile.cloneNode(true);
+	editFile.textContent = 'Edit file';
+	editFile.removeAttribute('data-ga-click');
+	editFile.href = url.assign({route: 'edit'}).toString();
+	select('[aria-label$="to make changes."]', dropdown)!.replaceWith(editFile);
+
+	// Fix the delete link
+	const deleteFile = editFile.cloneNode(true);
+	deleteFile.textContent = 'Delete file';
+	deleteFile.classList.add('menu-item-danger');
+	deleteFile.href = url.assign({route: 'delete'}).toString();
+	select('[aria-label$="delete this file."]', dropdown)!.replaceWith(deleteFile);
+}
+
+function init(): void {
+	const handleMenuOpening = pageDetect.isCompare() ? handleCompareMenuOpening : handlePRMenuOpening;
+	delegate(document, '.file-header:not([data-file-deleted="true"]) .js-file-header-dropdown > summary:not(.rgh-actionable-link)', 'click', handleMenuOpening);
+}
+
+features.add({
+	id: __filebasename,
+	description: 'Points the "View file" on compare view pages to the branch instead of the commit, so the Edit/Delete buttons will be enabled on the "View file" page, if needed.',
+	screenshot: 'https://user-images.githubusercontent.com/1402241/69044026-c5b17d80-0a26-11ea-86ae-c95f89d3669a.png'
+}, {
+	include: [
+		pageDetect.isPRFiles,
+		pageDetect.isPRCommit
+	],
+	exclude: [
+		// Only enabled on Open/Draft PRs. Editing files doesn't make sense after a PR is closed/merged.
+		() => !select.exists('.gh-header-meta [title$="Open"], .gh-header-meta [title$="Draft"]'),
+		// If you're viewing changes from partial commits, ensure you're on the latest one.
+		() => select.exists('.js-commits-filtered') && !select.exists('[aria-label="You are viewing the latest commit"]')
+	],
+	init
+}, {
+	include: [
+		pageDetect.isCompare
+	],
+	exclude: [
+		// If you cant create a PR you are probably looking at a permalink.
+		() => !select.exists('#new_pull_request'),
+		// Exclude repositories you don't have access to.
+		() => !select.exists([
```

Comment:
Can be merged

Reply:
The truth is that the first one is not needed at all I just did not know how to word it correctly without putting it into 2.
If `[name="collab_privs"]` does not exist then `'#new_pull_request'` does not exist.

I will give it a shot, I will try to be as clear as can be.

Reply:
What if it just looks for the "Create PR" button if that's the logic you're following? Could that work?

e.g. if `.existing-pull-button, [data-ga-click*="text:Create pull request"]` exists, then the links can be _fixed_

- ✅ FIX: PR exists: https://github.com/sindresorhus/refined-github/compare/master...yakov116:use-filePathFromSearch
- ✅ FIX: Can create PR: https://github.com/sindresorhus/refined-github/compare/a@b...yakov116:use-filePathFromSearch
- ❌ LEAVE: Cannot create PR: https://github.com/sindresorhus/refined-github/compare/a@b...yakov116:123

Reply:
That's fine with me. I though you wanted more of a detection if you are able to edit it too.

Reply:
Edit buttons appear everywhere on the site whether you can edit it directly or you have to fork it first.

Reply:
Great I never took me one of that